### PR TITLE
Update calls to deprecated iceberg Transform and Transforms 's API

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Update calls to deprecated iceberg Transform and Transforms's API

Deprecated API whose usage is updated
1. org.apache.iceberg.transforms.Transforms#identity, #bucket, #truncate, #year, #month, #day, #hour
2. org.apache.iceberg.transforms.Transform#apply, #toHumanString

## Release notes

(X) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by krvikash** [comment]
CI hit https://github.com/trinodb/trino/issues/15293

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java [hunk 1]
```diff
@@ -360,16 +361,17 @@ private static Optional<PartitionData> getPartitionData(List<PartitionColumn> co
             PartitionColumn column = columns.get(i);
             Block block = page.getBlock(column.getSourceChannel());
             Type type = column.getSourceType();
+            org.apache.iceberg.types.Type icebergType = toIcebergType(type);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java:364]: Can we get the Iceberg type from existing fields? e.g. `Schema outputSchema`

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java (hunk 1 above)
Comment (krvikash) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java]: Yes, After making `getPartitionData` a non-static method, we can use `outputSchema` to get the iceberg type.
``` java
 org.apache.iceberg.types.Type icebergType = outputSchema.findType(column.getField().sourceId());
````

**Comment by krvikash** [comment]
Thanks @ebyhr for merge :)

**Pull request merged**